### PR TITLE
[Rogue] T13/Phase 4 additions, APL tuning, Combat reforging defaults

### PR DESCRIPTION
**Pull request by TheBackstabi**
Dagger implementation mirrors that of SimC/Shadowcraft. Likely more APL adjustments that can be made, but ATM I'm being very conservative about additions that could impact T12 sims.

Default gearset selections remain on P3, plan to remain that way until P4 is imminent.

### CODE REVIEW

File: sim/rogue/assassination/assassination_test.go [hunk 1]
```diff
@@ -22,7 +22,7 @@ func TestAssassination(t *testing.T) {
 		OtherGearSets: []core.GearSetCombo{
 			core.GetGearSet("../../../ui/rogue/assassination/gear_sets", "p3_assassination"),
 			core.GetGearSet("../../../ui/rogue/assassination/gear_sets", "preraid_assassination"),
```
Comment (1337LutZ) [inline sim/rogue/assassination/assassination_test.go:24]: We don't really need to run tests for all sets unless they are explicitly testing individual things ofc. Can Preraid be removed?

File: ui/rogue/combat/sim.ts [hunk 2]
```diff
@@ -64,7 +65,7 @@ const SPEC_CONFIG = registerSpecConfig(Spec.SpecCombatRogue, {
 			// Running just under spell cap is typically preferrable to being over.
 			const spellHitSoftCapConfig = StatCap.fromPseudoStat(PseudoStat.PseudoStatSpellHitPercent, {
 				breakpoints: [16.95, 16.96, 16.97, 16.98, 16.99, 17],
-				capType: StatCapType.TypeSoftCap,
+				capType: StatCapType.TypeThreshold,
```
Comment (1337LutZ) [inline ui/rogue/combat/sim.ts:68]: Threshold will always try to get the highest possible cap

File: ui/rogue/combat/sim.ts [hunk 3]
```diff
@@ -64,7 +65,7 @@ const SPEC_CONFIG = registerSpecConfig(Spec.SpecCombatRogue, {
...
 				postCapEPs: [0, 0, 0, 0, 0, 0],
```
Comment (1337LutZ) [inline ui/rogue/combat/sim.ts:69]: Threshold only supports `[0]` a single array entry

File: ui/rogue/combat/sim.ts (hunk 2 above)
Comment (TheBackstabi) [reply ui/rogue/combat/sim.ts:68]: Not gonna lie, forgot I had switched this just to see what it did. Gonna put it back + have a few other changes in a bit.

**Comment by TheBackstabi** [comment]
Still working on this. Caught something last week so I've been in and out of sleep and haven't been able to focus on this

**Comment by TheBackstabi** [comment]
Alright pretty sure I've figured out everything I need to for Combat, everything should be good to go here.

**Review by 1337LutZ — APPROVED** [review]

**Pull request merged**